spec/lib/monitoring/middleware/prometheus_exporter_spec.rb

```diff
@@ -25,8 +25,8 @@
     it 'returns the app response' do
       get '/foo'
 
-      expect(last_response).to be_ok
-      expect(last_response.body).to eql('OK')
+      expect(response.status).to eql(200)
```

Comment:
Purely out of curiosity, how does `get` know to direct traffic to the `app` you defined. I haven't been able to find docs on this. I typically expect request specs to just work against the Rails application (minus the rack middleware).

Reply:
This may be a bit of misuse of a request spec, but to my understanding it does factor in the middleware layer. In this case the app gets defined by the lambda which is used to initialize to a new instance of the middleware (described_class).
`app = ->(_) { [200, { 'Content-Type' => 'text/html' }, ['OK']] }`
`described_class.new(app, **options)`

Requests make their way to the `call` method in the described class with the passed in lambda as the expected app response.

I'm pushing up a different version of the tests which I think is made a bit clearer by calling the middleware directly instead of leaning on request spec features. Plus it seems to be more inline with conventions I have seen for testing middleware. Let me know what you think.